python/tk_framework_desktopserver/shotgun/scripts/execute_command.py

```diff
@@ -59,67 +52,87 @@ def core_info(engine):
-        logger.info("")
-        logger.info(
-            "Your core API is located in <code>%s</code> and is shared with other "
+        engine.log_info(
+            "Your core API is located in `%s` and is shared with other "
             "projects." % install_root
         )
 
-    logger.info("")
     status = installer.get_update_status()
 
     if status == TankCoreUpdater.UP_TO_DATE:
-        logger.info(
-            "<b>You are up to date! There is no need to update the Toolkit "
-            "Core API at this time!</b>"
+        engine.log_info(
+            "*You are up to date! There is no need to update the Toolkit "
+            "Core API at this time!*"
         )
     elif status == TankCoreUpdater.UPDATE_BLOCKED_BY_SG:
         req_sg = installer.get_required_sg_version_for_update()
-        logger.warning(
-            "<b>A new version (%s) of the core API is available however "
-            "it requires a more recent version (%s) of Shotgun!</b>" % (lv, req_sg)
+        engine.log_warning(
+            "*A new version (%s) of the core API is available however "
+            "it requires a more recent version (%s) of Shotgun!*" % (lv, req_sg)
         )
     elif status == TankCoreUpdater.UPDATE_POSSIBLE:
         (summary, url) = installer.get_release_notes()
 
-        logger.info("<b>A new version of the Toolkit API (%s) is available!</b>" % lv)
-        logger.info("")
-        logger.info(
-            "<b>Change Summary:</b> %s <a href='%s' target=_new>"
-            "Click for detailed Release Notes</a>" % (summary, url)
+        engine.log_info("*A new version of the Toolkit API (%s) is available!*" % lv)
+        engine.log_info(
+            "*Change Summary:* %s [Click for detailed Release Notes](%s)" % (summary, url)
         )
-        logger.info("")
-        logger.info("In order to upgrade, execute the following command in a shell:")
-        logger.info("")
+        engine.log_info("In order to upgrade, execute the following command in a shell:")
 
         if sys.platform == "win32":
             tank_cmd = os.path.join(install_root, "tank.bat")
         else:
             tank_cmd = os.path.join(install_root, "tank")
 
-        logger.info("<code style='%s'>%s core</code>" % (code_css_block, tank_cmd))
-        logger.info("")
+        engine.log_info("*%s core*" % tank_cmd)
     else:
         raise sgtk.TankError("Unknown Upgrade state!")
 
+def pre_engine_start_callback(context):
+    """
+    The pre-engine-start callback that's given to the bootstrap API.
+    This callback handles attaching a custom logger to SGTK prior to
+    the Shotgun engine being initialized. This allows us to customize
+    the output of the logger in such a way that it is easily identified
+    and filtered before going back to the client for display.
+
+    :param context: The context object being used during the bootstrap
+        process.
+    """
+    import sgtk
```

Comment:
what core is this? guaranteed to be 0.18? or the core that we are bootstrapping into? please add comment